### PR TITLE
ARTEMIS-1364: Enable internal sorting in Hawtio web console

### PR DESCRIPTION
**Pull request by RaiSaurabh**
Enabled default sorting on the Hawtio web console.

### CODE REVIEW

**Comment by RaiSaurabh** [comment]
For sorting purpose AngularJS grid internal sorting should be enough for sorting columns. I checked it was not sorting on other tabs as well (Connections, Sessions, Consumers, Producers, Addresses and Queues). I tested the UI after the change and sorting works.

**Comment by michaelandrepearce** [comment]
@mtaylor i think we were looking at pushing sort down to broker as will be more elements than in FE table when many queues/address etc. Could you confirm?

**Comment by mtaylor** [comment]
@RaiSaurabh @michaelandrepearce Since the results are paged, we need to sort on the broker side.  If it doesn't work already then that's a bug on the broker that needs addressing.

**Comment by RaiSaurabh** [comment]
@mtaylor @michaelandrepearce The current implementation we set the sort column and option in the filter and pass it along to the Artemis server for the operation. But looking at the code on the Artemis server I see that it only check for the Filter components and not sorting. No such functionality is written for sorting of columns. With the default sorting operation provided by the Angular grid, I am able to sort the response table even on paging. Should not we go ahead with this?

**Comment by mtaylor** [comment]
@RaiSaurabh I am not sure how it can work with paging.  Since the full list needs to be sorted before the pages are populated.  This solution will receive the first N (page size) unsorted elements from a collection, then sort them, meaning only the page results are sorted.  We should instead implement this on the broker so that sorting + paging works.

**Comment by RaiSaurabh** [comment]
@mtaylor Ok. Thanks, I will start working on doing it in broker end. Will update once done.

**Comment by mtaylor** [comment]
@RaiSaurabh Awesome.  Thanks.

**Comment by RaiSaurabh** [comment]
Added the code to support sorting on the server side. Request @mtaylor to please review.

**Comment by RaiSaurabh** [comment]
Will fix and commit the indentation issue.

File: artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/ServerConsumerImpl.java [hunk 1]
```diff
@@ -1387,4 +1387,49 @@ public boolean isBrowsed() {
...
+
+   @Override
+   public SimpleString getQueueAddress() {
+       return getQueue().getAddress();
+   }
+
+   @Override
+   public String getSessionName() {
+       return server.getSessionByID(getSessionID()).getName();
+   }
```
Comment (mtaylor) [inline artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/ServerConsumerImpl.java]: You already have access to the session in the ServerConsumerImpl.

File: artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/ServerConsumerImpl.java [hunk 2]
```diff
@@ -1387,4 +1387,49 @@ public boolean isBrowsed() {
...
+
+   @Override
+   public String getConnectionClientID() {
+       return server.getSessionByID(getSessionID()).getRemotingConnection().getClientID();
+   }
```
Comment (mtaylor) [inline artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/ServerConsumerImpl.java]: Same comment as above and in all other places you are doing the lookup.

**Comment by RaiSaurabh** [comment]
@mtaylor This is strange the Checkstyle is not failing on my end but it is on the Jenkins.

**Comment by RaiSaurabh** [comment]
Got the CheckStyle validations. Will fix them and then commit and then squash.

**Comment by RaiSaurabh** [comment]
@mtaylor I have updated the code and squashed it into one commit. The build was successful.

**Comment by RaiSaurabh** [comment]
@mtaylor @michaelandrepearce Could you please merge the PR.

**Comment by mtaylor** [comment]
Nice work @RaiSaurabh.  Thanks a lot

**Pull request closed without merging**